File: m3-h2/src/main/java/ru/yegorpilipenko/otus/spring/m3h2/service/BookService.java
package ru.yegorpilipenko.otus.spring.m3h2.service;

import ru.yegorpilipenko.otus.spring.m3h2.dto.ShortBook;
import ru.yegorpilipenko.otus.spring.m3h2.model.Book;
import ru.yegorpilipenko.otus.spring.m3h2.model.Comment;

import java.util.Collection;
import java.util.Optional;

public interface BookService {

    Collection<ShortBook> getAll();

    Optional<Book> getById(String id);

    Book save(Book book);

    boolean removeById(String id);

    Book addComment(String bookId, Comment comment);

    Optional<Book> removeComment(String bookId, Comment comment);

}

File: m1-h5/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h5/service/BookService.kt
package ru.yegorpilipenko.otus.spring.m1h5.service

import ru.yegorpilipenko.otus.spring.m1h5.dto.ShortBook
import ru.yegorpilipenko.otus.spring.m1h5.model.Book

interface BookService {

    fun findAll(): Collection<ShortBook>

    fun findById(id: Long): Book?

    fun save(book: ShortBook, authorIds: Collection<Long>, genreIds: Collection<Long>): Book

    fun removeById(id: Long): Boolean

}
